File: src/transformers/features/http/HttpValidateQueryTransformer.ts
import { HttpValidateQueryProgrammer } from "../../../programmers/http/HttpValidateQueryProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace HttpValidateQueryTransformer {
    export const transform = GenericTransformer.scalar("http.validateQuery")(
        (project) => (modulo) =>
            HttpValidateQueryProgrammer.write(project)(modulo),
    );
}

File: packages/transform/src/features/http/HttpValidateQueryTransformer.go
package http

import (
	corehttp "github.com/samchon/typia/packages/core/src/programmers/http"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var HttpValidateQueryTransformer = httpValidateQueryTransformerNamespace{}

type httpValidateQueryTransformerNamespace struct{}

func (httpValidateQueryTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return corehttp.HttpValidateQueryProgrammer.Write(call.TypeText)
}
